testing/integration/container_cmd_test.go

```diff
@@ -112,22 +94,52 @@ func TestContainerCMD(t *testing.T) {
 	agentOutput := strings.Builder{}
 	cmd.Stderr = &agentOutput
 	cmd.Stdout = &agentOutput
-	cmd.Env = append(os.Environ(),
+	cmd.Env = append(os.Environ(), env...)
+	return cmd
+}
+
+func TestContainerCMD(t *testing.T) {
+	info := define.Require(t, define.Requirements{
+		Stack: &define.Stack{},
+		Local: false,
+		Sudo:  true,
+		OS: []define.OS{
+			{Type: define.Linux},
+		},
+		Group: "container",
+	})
+
+	ctx, cancel := context.WithTimeout(context.Background(), 1*time.Minute)
+	defer cancel()
+
+	agentFixture, err := define.NewFixtureFromLocalBuild(t, define.Version())
+	require.NoError(t, err)
+
+	fleetURL, err := fleettools.DefaultURL(ctx, info.KibanaClient)
+	if err != nil {
+		t.Fatalf("could not get Fleet URL: %s", err)
+	}
+
+	enrollmentToken := createPolicy(t, ctx, agentFixture, info)
+	env := []string{
 		"FLEET_ENROLL=1",
-		"FLEET_URL="+fleetURL,
-		"FLEET_ENROLLMENT_TOKEN="+enrollmentToken.APIKey,
+		"FLEET_URL=" + fleetURL,
+		"FLEET_ENROLLMENT_TOKEN=" + enrollmentToken,
 		// As the agent isn't built for a container, it's upgradable, triggering
 		// the start of the upgrade watcher. If `STATE_PATH` isn't set, the
 		// upgrade watcher will commence from a different path within the
 		// container, distinct from the current execution path.
-		"STATE_PATH="+agentFixture.WorkDir(),
-	)
+		"STATE_PATH=" + agentFixture.WorkDir(),
+	}
 
+	cmd := prepareContainerCMD(t, ctx, agentFixture, info, env)
 	t.Logf(">> running binary with: %v", cmd.Args)
 	if err := cmd.Start(); err != nil {
 		t.Fatalf("error running container cmd: %s", err)
 	}
 
+	agentOutput := cmd.Stderr.(*strings.Builder)
```

Comment:
I would prefer that the `strings.Builder` pointer is returned as a separate value by `prepareContainerCMD` instead of a direct cast here

Reply:
It's a test, I didn't want to add returned values not every test is gonna use. If I end up making changes, I might add it as well ;)